### PR TITLE
[2848] Provide a common ShareRepresentationModal

### LINKED ISSUE #2848: Provide a common ShareRepresentationDialog
Most representation kinds have (or should/will have) a "share" button in their toolbar, which opens a modal dialog with the direct URL of the representation and copies it into the system clipboard.

* Diagrams have [`ShareDiagramDialog`](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/ShareDiagramDialog.tsx)
* Gantt have [`ShareGanttModal.tsx`](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/gantt/frontend/sirius-components-gantt/src/share-gantt/ShareGanttModal.tsx) which is very similar but takes the URL to share as a prop instead of hard-coding the use of `window.location.href`.
* Portals and Deck representation will need one.
* Forms should also probably have one, but they currently do not have a toolbar where to place it.

Anyway, there is nothing specific to the nature of the representation shared in this modal, so it would be better to have a single common implementation.

In addition, the use of `window.location.href` (either hard-coded or passed as a prop) will soon become incorrect. For representations embedded inside a portal, sharing the representation should not share the current URL, which is the portal's URL.

Proposed interface for the shared implementation:

```js
export interface ShareRepresentationModalProps {
  editingContextId: string;
  representationId: string;
  onClose: () => void;
}
```

The URL shared would need to be computed from `editingContextId` and `representationId`.
The issue is that this mapping between `editingContextId` and `representationId` on one side and the URL on the other side is defined in `sirius-web-application/src/router/Router.tsx` at the application level.

### PR DESCRIPTION
**Pull request by pcdavid**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/2848

### CODE REVIEW

**Comment by pcdavid** [comment]
The build is red because of a single Cypress test failure, which is unrelated ("ResizeObserver loop limit exceeded").

File: doc/adrs/124_common_share_representation_modal.adoc
Comment (sbegaudeau) [inline doc/adrs/124_common_share_representation_modal.adoc]: `editingContextId` -> `representationId`

File: packages/gantt/frontend/sirius-components-gantt/src/Gantt.tsx
Comment (sbegaudeau) [inline packages/gantt/frontend/sirius-components-gantt/src/Gantt.tsx]: Is that a duplicate of `src/representation/Gantt.tsx` ? I'll check

File: doc/adrs/124_common_share_representation_modal.adoc
Comment (sbegaudeau) [reply doc/adrs/124_common_share_representation_modal.adoc]: Done

File: packages/gantt/frontend/sirius-components-gantt/src/Gantt.tsx
Comment (sbegaudeau) [reply packages/gantt/frontend/sirius-components-gantt/src/Gantt.tsx]: I think it's just a small detail from the rebase conflict that can be removed.

File: packages/portals/frontend/sirius-components-portals/src/representations/PortalToolbar.tsx
Comment (sbegaudeau) [inline packages/portals/frontend/sirius-components-portals/src/representations/PortalToolbar.tsx:50]: It seems that you have duplicated the whole portal toolbar, every button is present twice now.

Comment (sbegaudeau) [reply packages/portals/frontend/sirius-components-portals/src/representations/PortalToolbar.tsx:50]: I've fixed it

**Comment by sbegaudeau** [comment]
There seems to be an issue with the commit message or a bug with Github because it looks very odd:

Capture d’écran 2024-01-24 à 10 07 23

I'll fix it too. I'm investigating one last issue, once it will be done I'll merge it.

File: packages/portals/frontend/sirius-components-portals/src/representations/PortalToolbar.tsx
Comment (pcdavid) [reply packages/portals/frontend/sirius-components-portals/src/representations/PortalToolbar.tsx:50]: Probably a mistake during yesterday's rebase, sorry.

**Comment by pcdavid** [comment]
> There seems to be an issue with the commit message or a bug with Github because it looks very odd

Looks like I amended the then-current master when I rebased yesterday.

**Comment by sbegaudeau** [comment]
> > There seems to be an issue with the commit message or a bug with Github because it looks very odd
> 
> Looks like I amended the then-current master when I rebased yesterday.

That may be it indeed, I was very surprised to see that 😛, I'll wait for the build and merge the PR this morning

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**
